doc/monads/readers.lean

```diff
@@ -0,0 +1,196 @@
+/-!
+# Readers
+
+In the [previous section](monads.lean.md) you learned about the conceptual idea of monads. You learned
+what they are, and saw how some common types like `IO` and `Option` work as monads. Now in this
+part, you will be looking at some other useful monads. In particular, the `ReaderM` monad.
+
+## How to do Global Variables in Lean?
+
+In Lean, your code is generally "pure", meaning functions can only interact with the arguments
+passed to them. This effectively means you cannot have global variables. You can have global
+definitions, but these are fixed at compile time. If some user behavior might change them, you have
+to wrap them in the `IO` monad, which means they can't be used from pure code.
+
+Consider this example. Here, you want to have an `Environment` containing different parameters as a
+global variable. However, you want to load these parameters from the process environment variables,
+which requires the `IO` monad.
+-/
+
+structure Environment where
+  path : String
+  home : String
+  user : String
+  deriving Repr
+
+def getEnvDefault (name : String): IO String := do
+  pure <| match (← IO.getEnv name) with
```

Comment:
I think you did not introduce the nested monadic action notation `(<- ...)` yet.

Reply:
Perhaps, write it as follows at this point.
```
def getEnvDefault (name : String) : IO String := do
 let val? ← IO.getEnv name
 match val? with
 | none => pure ""
 | some s => pure s
```

Reply:
Fixed in https://github.com/leanprover/lean4/pull/1505/commits/6fed17209a17e523b02d2dd6b0cc773ec551738f